GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -0,0 +1,578 @@
+            PhysicalFileSystem fileSystem,
+            bool dryRun,
+            bool noVerify,
+            out NuGetUpgrader nuGetUpgrader,
+            out bool isConfigured,
+            out string error)
+        {
+            NuGetUpgraderConfig upgraderConfig = new NuGetUpgraderConfig(tracer, new LocalGVFSConfig());
+            nuGetUpgrader = null;
+            isConfigured = false;
+
+            if (!upgraderConfig.TryLoad(out error))
+            {
+                nuGetUpgrader = null;
+                return false;
+            }
+
+            if (!(isConfigured = upgraderConfig.IsConfigured(out error)))
+            {
+                return false;
+            }
+
+            // At this point, we have determined that the system is set up to use
+            // the NuGetUpgrader
+
+            if (!upgraderConfig.IsReady(out error))
+            {
+                return false;
+            }
+
+            string authUrl;
+            if (!TryCreateAzDevOrgUrlFromPackageFeedUrl(upgraderConfig.FeedUrl, out authUrl, out error))
+            {
+                return false;
+            }
+
+            if (!TryGetPersonalAccessToken(
+                    GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath(),
+                    authUrl,
+                    tracer,
+                    out string token,
+                    out string getPatError))
+            {
+                error = $"NuGetUpgrader was not able to acquire Personal Access Token to access NuGet feed. Error: {getPatError}";
+                tracer.RelatedError(error);
+                return false;
+            }
+
+            nuGetUpgrader = new NuGetUpgrader(
+                ProcessHelper.GetCurrentProcessVersion(),
+                tracer,
+                fileSystem,
+                dryRun,
+                noVerify,
+                upgraderConfig,
+                ProductUpgraderInfo.GetAssetDownloadsPath(),
+                token);
+
+            return true;
+        }
+
+        public static bool TryCreateAzDevOrgUrlFromPackageFeedUrl(string packageFeedUrl, out string azureDevOpsUrl, out string error)
+        {
+            // We expect a URL of the form https://pkgs.dev.azure.com/{org}
+            // and want to convert it to a URL of the form https://{org}.visualstudio.com
+            Regex packageUrlRegex = new Regex(
+                @"^https://pkgs.dev.azure.com/(?<org>.+?)/",
+                RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
+            Match urlMatch = packageUrlRegex.Match(packageFeedUrl);
+
+            if (!urlMatch.Success)
+            {
+                azureDevOpsUrl = null;
+                error = $"Input URL {packageFeedUrl} did not match expected format for an Azure DevOps Package Feed URL";
+                return false;
+            }
+
+            string org = urlMatch.Groups["org"].Value;
+
+            azureDevOpsUrl = urlMatch.Result($"https://{org}.visualstudio.com");
```

Comment:
I think you can drop `string org = urlMatch.Groups["org"].Value` and just do: `azureDevOpsUrl = urlMatch.Result("https://${org}.visualstudio.com");` since you already have a named capture group.

Reply:
Also, this does assume that the user has cloned with the old URLs. Should this still be configurable to ensure both URL styles can be used?